spirv_msl.cpp

```diff
@@ -14817,10 +14866,52 @@ void CompilerMSL::analyze_argument_buffers()
 		});
 
 		uint32_t member_index = 0;
+		uint32_t consumed_arg_slots = 0;
 		for (auto &resource : resources)
 		{
 			auto &var = *resource.var;
 			auto &type = get_variable_data_type(var);
+
+			// If needed, synthesize and add a padding member
+			if (msl_options.pad_argument_buffer_resources)
+			{
+				uint32_t pad_cnt = resource.index - consumed_arg_slots;
+				if (pad_cnt)
+				{
+					string mbr_name = join("_m", member_index, "_pad");
+					set_member_name(buffer_type.self, member_index, mbr_name);
+
+					bool use_array = pad_cnt > 1;
+					uint32_t pad_type_id = ir.increase_bound_by(use_array ? 2 : 1);
+					auto &pad_type = set<SPIRType>(pad_type_id);
+					pad_type.basetype = SPIRType::UInt64;
```

Comment:
This is questionable. Doesn't this basically assume that descriptors take up 8 bytes? I would normally expect this kind of interface to specify the descriptor type used for specific IDs.

Reply:
AMD texture descriptors are 16 or 32 bytes for example.